src/beta/resources/referral.js

```diff
@@ -0,0 +1,137 @@
+     * @returns {Promise<boolean>}
+     */
+    static async updateEmail(referralUserId, email) {
+      try {
+        const newParams = { user: { email } };
+        await api.put(`${this._url}/${referralUserId}`, newParams);
+        return true;
+      } catch (e) {
+        return Promise.reject(e);
+      }
+    }
+
+    /**
+     * Add a credit card to the referral's account
+     * @param {string} referralApiKey - The referral user's production API key.
+     * @param {object} creditCardDetails  - Credit card details:
+     * @param {string} creditCardDetails.number  - Credit card number
+     * @param {string} creditCardDetails.expirationMonth  - Credit card expiration month
+     * @param {string} creditCardDetails.expirationYear  - Credit card expiration year
+     * @param {string} creditCardDetails.cvc  - Credit card CVC
+     * @param {string} primaryOrSecondary  - Whether to add the card as the primary or secondary card (defaults to primary)
+     * @returns {Promise<never>}
+     */
+    static async addCreditCard(
+      referralApiKey,
+      creditCardDetails = {},
+      primaryOrSecondary = 'primary',
+    ) {
+      const stripeCreditCardId = await Referral.#sendCardDetailsToStripe(creditCardDetails);
+      const paymentMethod = await Referral.#sendCardDetailsToEasyPost(
+        referralApiKey,
+        stripeCreditCardId,
+        primaryOrSecondary,
+      );
+      return paymentMethod;
+    }
+
+    static async #getStripeKey() {
+      const response = await api.get('partners/stripe_public_key');
+      return response.body.public_key;
+    }
+
+    /*
+     * Send the credit card details to Stripe to get a Stripe credit card ID.
+     * @param {object} creditCardDetails  - Credit card details:
+     * @param {string} creditCardDetails.number  - Credit card number
+     * @param {string} creditCardDetails.expirationMonth  - Credit card expiration month
+     * @param {string} creditCardDetails.expirationYear  - Credit card expiration year
+     * @param {string} creditCardDetails.cvc  - Credit card CVC
+     * @returns {Promise<string>} - Stripe credit card ID
+     */
+    static async #sendCardDetailsToStripe(creditCardDetails = {}) {
+      const { number, expirationMonth, expirationYear, cvc } = creditCardDetails;
+      const stripeKey = await Referral.#getStripeKey();
+
+      const req = superagent.post(this.#stripeCreditCardUrl).set({
+        Authorization: `Bearer ${stripeKey}`,
+        'Content-Type': 'application/x-www-form-urlencoded',
+      });
+      req.send({
+        card: {
+          number,
+          exp_month: expirationMonth,
+          exp_year: expirationYear,
+          cvc,
+        },
+      });
+      try {
+        const res = await req;
+        return res.body.id;
+      } catch (e) {
+        if (e.response && e.response.body) {
+          throw new RequestError(e.response.body, this.#stripeCreditCardUrl);
+        }
+        throw e;
```

Comment:
Let's be explicit over implicit. `res` should be `response`, `req` should be `request`, `e` should be `error`